aten/src/ATen/native/CPUBlas.cpp

```diff
@@ -0,0 +1,250 @@
+#include <ATen/native/CPUBlas.h>
+#include <ATen/Config.h>
+
+#include <climits>
+
+#if AT_BUILD_WITH_BLAS()
+extern "C" void dgemm_(char *transa, char *transb, int *m, int *n, int *k, double *alpha, const double *a, int *lda, const double *b, int *ldb, double *beta, double *c, int *ldc);
+extern "C" void sgemm_(char *transa, char *transb, int *m, int *n, int *k, float *alpha, const float *a, int *lda, const float *b, int *ldb, float *beta, float *c, int *ldc);
+extern "C" void cgemm_(char *transa, char *transb, int *m, int *n, int *k, void *alpha, const void *a, int *lda, const void *b, int *ldb, void *beta, void *c, int *ldc);
+extern "C" void zgemm_(char *transa, char *transb, int *m, int *n, int *k, void *alpha, const void *a, int *lda, const void *b, int *ldb, void *beta, void *c, int *ldc);
+#endif  // AT_BUILD_WITH_BLAS()
+
+#ifdef USE_FBGEMM
+#include <fbgemm/FbgemmI64.h>
+#endif  // USE_FBGEMM
+
+namespace at {
+namespace native {
+namespace cpublas {
+namespace {
+
+void normalize_last_dims(
+    TransposeType transa, TransposeType transb,
+    int64_t m, int64_t n, int64_t k,
+    int64_t &lda, int64_t &ldb, int64_t &ldc) {
```

Comment:
It will be easier to see what this code does if you take in `int64_t*` instead of `int64_t&`, as it will be more evident which arguments constitute outputs.